### PR TITLE
Add linter for mixed use of assignment operators

### PR DESCRIPTION
**Pull request by ayazhafiz**

### CODE REVIEW

**Comment by ayazhafiz** [comment]
@lukebhan you will like this :)

File: libslide/src/linter/stmt/homogenous_assignment.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    ///a = 1
+    ///b := 1
+    ///```
+    ///
+    ///This can be misleading or confusing, as these two operators are syntactically different (and
+    ///semantically different in canonical mathematics notation), but are treated the same in slide.
+    ///
+    ///For this reason, it is suggested that exclusively `=` or `:=` are used for assignments in
+    ///slide programs.
+    L0004: HomogenousAssignmentLinter
```
Comment (lukebhan) [inline libslide/src/linter/stmt/homogenous_assignment.rs:17]: what is this: L0004?

**Review by lukebhan — APPROVED** [review]
this is so clean. I love how easy it is to write diagonistic with your span.

File: libslide/src/linter/stmt/homogenous_assignment.rs (hunk 1 above)
Comment (ayazhafiz) [reply libslide/src/linter/stmt/homogenous_assignment.rs:17]: it is the diagnostic code

**Pull request merged**